subprojects/language-native/src/main/java/org/gradle/language/plugins/NativeBasePlugin.java

```diff
@@ -351,13 +351,12 @@ private void addOutgoingConfigurationForLinkUsage(SoftwareComponentContainer com
         });
     }
 
-    private void addOutgoingConfigurationForRuntimeUsage(SoftwareComponentContainer components, final ConfigurationContainer configurations) {
+    private void addOutgoingConfigurationForRuntimeUsage(SoftwareComponentContainer components, final RoleBasedConfigurationContainerInternal configurations) {
         components.withType(ConfigurableComponentWithRuntimeUsage.class, component -> {
             Names names = component.getNames();
 
-            Configuration runtimeElements = configurations.create(names.withSuffix("runtimeElements"));
+            Configuration runtimeElements = configurations.consumable(names.withSuffix("runtimeElements"));
```

Comment:
Must be a consumable bucket. This applies in many places in this PR so I'll stop commenting on them in this PR, but we can't just disable declaration on a configuration without deprecating that usage first.